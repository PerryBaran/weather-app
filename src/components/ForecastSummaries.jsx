import React from "react";
import "../styles/ForecastSummaries.css";
import PropTypes from "prop-types";
import ForecastSummary from "./ForecastSummary";

function ForecastSummaries(props) {
  const { forecasts } = props;

  return (
    <div className="forecast-summaries">
      {forecasts.map((forecast) => {
        return <ForecastSummary forecast={forecast} key={forecast.date} />;
      })}
    </div>
  );
}

ForecastSummaries.propTypes = {
  forecasts: PropTypes.arrayOf(
    PropTypes.shape({
      date: PropTypes.number,
      description: PropTypes.string,
      icon: PropTypes.string,
      temperature: PropTypes.shape({
        max: PropTypes.number,
      }),
    })
  ).isRequired,
};

export default ForecastSummaries;
