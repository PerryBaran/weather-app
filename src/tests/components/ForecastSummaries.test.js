import React from "react";
import { render } from "@testing-library/react";
import ForecastSummaries from "../../components/ForecastSummaries";

describe("ForecastSummaries", () => {
  const validProps = [
    {
      date: 123,
      description: "Desc 1",
      icon: "Icon1",
      temperature: {
        max: 1,
      },
    },
    {
      date: 456,
      description: "Desc 2",
      icon: "Icon2",
      temperature: {
        max: 2,
      },
    },
  ];

  describe("snapshot", () => {
    it("renders correctlty", () => {
      const { asFragment } = render(
        <ForecastSummaries forecasts={validProps} />
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe("list", () => {
    const { getAllByTestId } = render(
      <ForecastSummaries forecasts={validProps} />
    );

    expect(getAllByTestId("forecast-summary")).toHaveLength(validProps.length);
  });
});
