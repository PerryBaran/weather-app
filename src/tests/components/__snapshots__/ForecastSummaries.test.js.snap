// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummaries snapshot renders correctlty 1`] = `
<DocumentFragment>
  <div
    class="forecast-summaries"
  >
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        123
      </div>
      <div
        class="forecast-summary__icon"
        data-testid="forecast-icon"
      >
        Icon1
      </div>
      <div
        class="forecast-summary__temperature"
      >
        1°C
      </div>
      <div
        class="forecast-summary__description"
      >
        Desc 1
      </div>
    </div>
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        456
      </div>
      <div
        class="forecast-summary__icon"
        data-testid="forecast-icon"
      >
        Icon2
      </div>
      <div
        class="forecast-summary__temperature"
      >
        2°C
      </div>
      <div
        class="forecast-summary__description"
      >
        Desc 2
      </div>
    </div>
  </div>
</DocumentFragment>
`;
